input/otlp/logs.go

```diff
@@ -239,3 +245,28 @@ func (c *Consumer) convertLogRecord(
 
 	return event
 }
+
+// https://www.elastic.co/guide/en/ecs/current/ecs-data_stream.html
+func sanitizeDataStream(k string) string {
+	k = strings.ToLower(k)
+	if strings.ContainsAny(k, DisallowedDataStreamRunes) {
+		k = strings.Map(replaceReservedRune, k)
+	}
+	// Cannot start with  _, +
+	// https://github.com/elastic/ecs/blob/main/rfcs/text/0009-data_stream-fields.md
+	if k[0] == '_' || k[0] == '+' {
+		k = k[1:]
+	}
+	if utf8.RuneCountInString(k) > MaxDataStreamRunes {
+		return string([]rune(k)[:MaxDataStreamRunes])
```

Comment:
whether it should be 100 unicode chars or 100 bytes depends on discussion in https://github.com/elastic/elasticsearch/issues/112936